catalyst-gateway/bin/src/db/index/queries/caches/assets/ada.rs

```diff
@@ -0,0 +1,160 @@
+//! Cache for TXO ADA Assets by Stake Address Queries
+use std::sync::Arc;
+
+use get_size2::GetSize;
+use moka::policy::EvictionPolicy;
+use tracing::{debug, error};
+
+use crate::{
+    db::{
+        index::queries::staked_ada::{
+            get_txo_by_stake_address::{GetTxoByStakeAddressQuery, GetTxoByStakeAddressQueryKey},
+            update_txo_spent::UpdateTxoSpentQueryParams,
+        },
+        types::DbStakeAddress,
+    },
+    metrics::caches::txo_assets::{txo_assets_hits_inc, txo_assets_misses_inc},
+    service::utilities::cache::Cache,
+    settings::Settings,
+};
+
+/// Cache for TXO ADA Assets by Stake Address Queries for Persistent Sessions
+pub(crate) struct AssetsAdaCache {
+    /// Interior cache type.
+    inner: Cache<DbStakeAddress, Arc<Vec<GetTxoByStakeAddressQuery>>>,
+}
+
+impl AssetsAdaCache {
+    /// Name for cache
+    const CACHE_NAME: &str = "Cardano UTXO ADA Assets Cache";
+
+    /// Function to determine the weighted size of cache entries.
+    ///
+    /// Returns the number of TXOs associated with a Stake Address.
+    fn weigher_fn(k: &DbStakeAddress, v: &Arc<Vec<GetTxoByStakeAddressQuery>>) -> u32 {
```

Comment:
What do you think to move this function under to the `Cache` type ? It looks like the definition is pretty much the same in all cases.